spec.html

```diff
@@ -30522,6 +30522,7 @@
             1. Let _index_ be _O_.[[ArrayIteratorNextIndex]].
             1. Let _itemKind_ be _O_.[[ArrayIterationKind]].
             1. If _a_ has a [[TypedArrayName]] internal slot, then
+              1. Perform ? ValidateTypedArray(_a_).
```

Comment:
It's unlikely that implementations would perform the redundant checks, but seems fair.